openupgradelib/openupgrade_160.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2022 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+"""This module provides simple tools for OpenUpgrade migration, specific for
+the 15.0 -> 16.0 migration.
+"""
+from .openupgrade import logged_query, table_exists
+from odoo.tools.translate import _get_translation_upgrade_queries
+
+
+def migrate_translations_to_jsonb(cr, field):
+    """
+    Execute queries returned by _get_translation_upgrade_queries
+    on table _ir_translation if exists, or on table ir_translation.
+
+    :param field: translatable field for which translations must be migrated
+    """
+    migrate_queries, cleanup_queries = _get_translation_upgrade_queries(
```

Comment:
IMO, The method should take the environment and a list of tuple(model name, field name) as parameter. 

```python
def migrate_translations_to_jsonb(env, fields_spec):
    for model, field_name in fields_spec:
        field = env[model]._fields[field_name]
        ....
```

To avoid list concatenation, you should use [`itertools.chain`](https://docs.python.org/3/library/itertools.html#itertools.chain)

```python
for query in intertools.chain(migrate_queries +  cleanup_queries):
    ....
```

The code could also be factorized.

```python
initial_translation_tables = None
if table_exists(cr, "_ir_translation"):
    # table renamed by Odoo's migration scripts
    initial_translation_table = "_ir_translation"
elif table_exists(cr, "ir_translation"):
    # Module migrated by OCA
    initial_translation_table = "ir_translation"
if initial_translation_table:
    for query = itertools.chain(_get_translation_upgrade_queries(cr, field))
         if initial_translation_table == 'ir_translation':
             query = query.replace("_ir_translation", "ir_translation")
         logged_query(cr, query)
```

Reply:
Done. Thanks for the comments.
I just used itertools.chain.from_iterable() instead of itertools.chain()